Library/Homebrew/cache_store.rb

```diff
@@ -62,7 +62,10 @@ def created?
   # @return [DBM] db
   def db
     # DBM::WRCREAT: Creates the database if it does not already exist
-    @db ||= DBM.open(dbm_file_path, DATABASE_MODE, DBM::WRCREAT)
+    @db ||= begin
+      HOMEBREW_CACHE.mkpath
```

Comment:
can this fail on a permissions exception?

Reply:
Theoretically but we assume it won't in enough other places I don't think it's worth special casing here.

Reply:
ok